packages/core/src/main/conversation/ConversationService.test.node.ts

```diff
@@ -70,17 +84,23 @@ describe('ConversationService', () => {
       it(`calls callbacks when sending '${payloadBundle.type}' message is starting and successful`, async () => {
         const conversationService = buildConversationService();
         const sentTime = new Date().toISOString();
+        const onStart = jasmine.createSpy().and.returnValue(Promise.resolve(true));
+        const onSuccess = jasmine.createSpy();
+
         spyOn<any>(conversationService, 'sendGenericMessage').and.returnValue(Promise.resolve({time: sentTime}));
-        const callbacks = {onStart: jasmine.createSpy(), onSuccess: jasmine.createSpy()};
+        // const onReconnect = jasmine.createSpy().and.returnValue(getServerAddress());
+
         const promise = conversationService.send({
```

Comment:
react testing library has waitFor, any reason for not using that?

Reply:
This is a library that doesn't use react (no frontend part), therefore we don't have react-testing-library.

Reply:
could also use the non-react version? not sure if it makes sense in this case